git-cleanup

```diff
@@ -36,7 +36,7 @@ while getopts nlrsvh flag; do
 done
 shift $(($OPTIND - 1))
 
-if [ $remotes -eq 0 -a $locals -eq 0 ]; then
+if [ $remotes -eq 0 ] && [ $locals -eq 0 ]; then
```

Comment:
Why make this change? The `-a` flag works everywhere, in `/bin/sh` and `/bin/bash`. I believe the `&&` approach will only work in systems with `/bin/bash`, so I don't really see the benefit here.

Reply:
From personal experience, this is detected by [ShellCheck SC2107](https://www.shellcheck.net/wiki/SC2107), and one of the sources they cite is [BashPitfalls 6](https://mywiki.wooledge.org/BashPitfalls#pf6):

> The binary -a and -o, and ( / ) (grouping) operators are XSI extensions to the POSIX standard. All are marked as obsolescent in POSIX-2008. They should not be used in new code.

Reply:
Oh, I see. I think I mis-read the original diff and thought it "just" changed the `-a` to `&&` like`if [ $remotes -eq 0 && $locals -eq 0 ]`, as if the `&&` was an argument to `test` (aka `[`). Now I see it just splits the command into two separate `[` commands, with `&&` being normal shell operators to combine them. That's great actually! Thanks for pointing to these resources 👍

Reply:
Thank you @harkabeeparolus for adding the extra detail! Just got back to this today